lang/src/accounts/account_loader.rs

```diff
@@ -236,21 +236,16 @@ impl<'info, T: ZeroCopy + Owner> Accounts<'info> for AccountLoader<'info, T> {
 impl<'info, T: ZeroCopy + Owner> AccountsExit<'info> for AccountLoader<'info, T> {
     // The account *cannot* be loaded when this is called.
     fn exit(&self, _program_id: &Pubkey) -> Result<()> {
-        let mut data = self.acc_info.try_borrow_mut_data()?;
-        let dst: &mut [u8] = &mut data;
-        let mut writer = BpfWriter::new(dst);
-        writer.write_all(&T::discriminator()).unwrap();
+        if !crate::common::is_closed(&self.acc_info) {
```

Comment:
The only reason this would be triggered is if `.close()` is called manually within the instruction, or the account is closed within an internal CPI. Without this check, closing an `AccountLoader` within an ix would result in the `AccountsExit` failing at the end if the account is marked as `mut`.

Reply:
It's weird right though that it doesn't already have the same check `Account` namely it doesn't have the same
```
if &T::owner() == program_id {
```
check.

Reply:
But yes, I get you. It seems like maybe it was an oversight that this `exit` function doesn't have that program id owner check in the first place. Looks good to me then, though maybe it's worth adding in the program owner check?

Reply:
If there wasn't a reason to not include that additional check before, then sure. I wasn't sure if that was an intentional decision to not include it. Not too familiar with the zero copy loaders

Reply:
Yea sounds unintentional although if that check is false the transaction will abort.

Reply:
If I'm understanding correctly, we should add the same checks I'm adding to Account on to AccountLoader?

Reply:
yes, i think so

Reply:
Gotcha. Adding those in and hopefully we should be ready to merge in?